cumulus_library/studies/core/core_templates/condition.sql.jinja

```diff
@@ -0,0 +1,51 @@
+{% import 'core_utils.jinja' as utils %}
+
+CREATE TABLE core__condition AS
+WITH temp_condition AS (
+    SELECT
+        concat('Condition/', c.id) AS condition_ref,
+        {{- utils.basic_nullable_cols(
+            'condition',
+            'c',
+            ['id','category','verificationStatus','clinicalStatus'], schema) 
+        }},
+        {{- utils.depth_one_nullable_cols(
+                'condition',
+                'c',
+                [
+                    ('subject', 'reference', 'subject_ref'),
+                    ('encounter', 'reference', 'encounter_ref')
+                ], 
```

Comment:
nit: there is a world where you can combine basic_nullable_cols and depth_one_nullable_cols into a generic nullable_cols method.

Accept a list of str | tuple (jinja has type inspection, so your method could tell the difference). If a string, split it by `.` and use that as the depth indicators. If a tuple, take two strings, the left is the field path, the right is the new alias.

So together, these two calls could look like:

```
        {{- utils.nullable_cols(
                'condition',
                'c',
                [
                    'id', 'category', 'verificationStatus', 'clinicalStatus',
                    ('subject.reference', 'subject_ref'),
                    ('encounter.reference', 'encounter_ref'),
                ], 
          )}}
```

(I'm only proposing that because I can already foresee the need for `depth_two_nullable_cols` and it's feeling a bit wobbly.)

Reply:
so i had already planned for a two depth call in the future, thinking that separation would be easier maintenance wise - but if you're feeling that type inference is ok at this complexity, you can talk me out of it.

Reply:
I don't feel strongly about this - I just have a personal bias towards high-level API vs low-level API. :smile: but the tradeoffs around that are squarely on you, so whatever you like.

Speaking of, I might want to use these schema concepts from the quality study. In my head, using stuff from Python feels OK, but not jinja utilities (so I'd have to recreate these macros myself). Does that vibe with you? Or am I not even supposed to be calling into Python... - a future task could be to clarify which API is public and which isn't (maybe using my `__init__.py` vibes of putting public API in there).

Reply:
so i'm not quite there yet (especially not in one of these big PRs), but I like the idea of some sort of global location for jinja macros that could be imported in other studies. I think the trick is deciding what is study specific, and what is truly global. 

In my head, this was core specific, since I'm not expecting study authors to routinely need to engage with the nuances of nested FHIR elements. I think the quality study is an exception to the rule since you're interacting with base FHIR there - so, maybe these could be global, but in like a `fhir_utils` type bucket?

Reply:
I don't want to burden these early macros with future thinking - I'm happy to re-implement / copy some of these schema macros.

The only thing I'd ask is that moving some of the schema python stuff around would be considered a breaking change. Is that a fair burden?

Reply:
Yeah I think so - I want to leave this comment open specifically because i might make that as part of the breaking changes this is incurring once the other tables are cut over.